src/main/java/com/timgroup/statsd/StatsDProcessor.java

```diff
@@ -176,13 +139,22 @@ public Telemetry getTelemetry() {
         return telemetry;
     }
 
-    boolean isShutdown() {
-        return shutdown;
-    }
-
     void shutdown() {
         shutdown = true;
         aggregator.stop();
-        executor.shutdown();
+        for (int i = 0 ; i < workers.length ; i++) {
+            workers[i].interrupt();
```

Comment:
Should the worker also join here?

Reply:
A countdown-latch is used to detect when all the workers have finished (this is the same as before, except the logic has been tweaked to avoid needing an intermediate thread pool to start the workers and wait for them.) This could be changed to use `join` but we'd still want to signal all workers first and then wait for them in a separate call rather than signal one and wait for it before going onto the next.